File: test/fixtures/ruby/post_form.rb
require 'net/http'

uri = URI('http://localhost:28139/post-to-me.php')
req = Net::HTTP::Post.new(uri)

req.set_form(
  [
    [
      'username',
      'davidwalsh'
    ],
    [
      'password',
      'something'
    ]
  ],
  'multipart/form-data'
)

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/go/post_form.go
package main

import (
	"bytes"
	"fmt"
	"io"
	"log"
	"mime/multipart"
	"net/http"
)

func main() {
	form := new(bytes.Buffer)
	writer := multipart.NewWriter(form)
	formField, err := writer.CreateFormField("username")
	if err != nil {
		log.Fatal(err)
	}
	_, err = formField.Write([]byte("davidwalsh"))

	formField, err = writer.CreateFormField("password")
	if err != nil {
		log.Fatal(err)
	}
	_, err = formField.Write([]byte("something"))

	writer.Close()

	client := &http.Client{}
	req, err := http.NewRequest("POST", "http://localhost:28139/post-to-me.php", form)
	if err != nil {
		log.Fatal(err)
	}
	req.Header.Set("Content-Type", writer.FormDataContentType())
	resp, err := client.Do(req)
	if err != nil {
		log.Fatal(err)
	}
	defer resp.Body.Close()
	bodyText, err := io.ReadAll(resp.Body)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("%s\n", bodyText)
}
